glibc warning: _BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE

Compiling jq 1.6 with glibc >=2.20 (2.20 released 5 years ago!) gives a warning:

```
In file included from jq/src/builtin.c:10:
In file included from /usr/include/sys/time.h:21:
/usr/include/features.h:184:3: warning: "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
```

Relevant piece from glibc 2.20 [release notes](https://sourceware.org/ml/libc-alpha/2014-09/msg00088.html):

```
* The _BSD_SOURCE and _SVID_SOURCE feature test macros are no longer
  supported; they now act the same as _DEFAULT_SOURCE (but generate a
  warning). 
```

--- a/src/builtin.c
+++ b/src/builtin.c
@@ -1,5 +1,6 @@
-#define _BSD_SOURCE
-#define _GNU_SOURCE
+#ifndef _GNU_SOURCE
+# define _GNU_SOURCE
+#endif
 #ifndef __sun__
 # define _XOPEN_SOURCE
 # define _XOPEN_SOURCE_EXTENDED 1

